sanitizers/app/src/main/cpp/CMakeLists.txt

```diff
@@ -0,0 +1,86 @@
+# For more information about using CMake with Android Studio, read the
+# documentation: https://d.android.com/studio/projects/add-native-code.html
+
+# Sets the minimum version of CMake required to build the native library.
+
+cmake_minimum_required(VERSION 3.18.1)
+
+# Declares and names the project.
+
+project("sanitizers")
+
+# Creates and names a library, sets it as either STATIC
+# or SHARED, and provides the relative paths to its source code.
+# You can define multiple libraries, and CMake builds them for you.
+# Gradle automatically packages shared libraries with your APK.
+
+add_library( # Sets the name of the library.
+        sanitizers
+
+        # Sets the library as a shared library.
+        SHARED
+
+        # Provides a relative path to your source file(s).
+        native-lib.cpp)
+
+if(SANITIZE)
+if(${SANITIZE} STREQUAL "hwasan")
+    message("Using hwasan")
+    target_compile_options(sanitizers PUBLIC -fsanitize=hwaddress -fno-omit-frame-pointer)
+    set_target_properties(sanitizers PROPERTIES LINK_FLAGS -fsanitize=hwaddress)
```

Comment:
You should be able to use https://cmake.org/cmake/help/latest/command/target_link_options.html these days. We have 3.22 available.

Reply:
Done.